CRM/Utils/Check/Component/Env.php

```diff
@@ -66,13 +66,21 @@ public function checkPhpVersion() {
       );
     }
     else {
+      $date = '';
+      $dateFormat = Civi::Settings()->get('dateformatshortdate');
+      if (version_compare(phpversion(), 5.4) < 0) {
+        $date = CRM_Utils_Date::customFormat('2017-12-31', $dateFormat);
+      }
+      elseif (version_compare(phpversion(), 5.5) < 0) {
+        $date = CRM_Utils_Date::customFormat('2018-02-28', $dateFormat);
+      }
       $messages[] = new CRM_Utils_Check_Message(
         __FUNCTION__,
-        ts('This system uses PHP version %1. CiviCRM can be installed on this version, but some specific features are known to fail or degrade. Version %3 is the bare minimum to avoid known issues, and version %2 is recommended.',
+        ts('This system uses PHP version %1. CiviCRM can be installed on this version, However PHP version %1, will not work in releases published after %2, and version %3 is recommended.',
```

Comment:
Comments on grammar: in the message `However PHP version %1, will not work`, the comma separates the subject ("PHP version %1") and the predicate ("will not work..."). That's weird. Either of these formulations would be more acceptable:
 * Use the comma to set aside the interjection ("However, PHP version %1 will not work...") (*my personal preference*)
 * Kill the comma ("However PHP version %1 will not work...") (*may be OK since "However" is only one word*)